CreateMemSet: llvm10: Fix compilation errors

ovpanait opened the pull request:
The following llvm commit:
"[Alignment][NFC] CreateMemSet use MaybeAlign" (https://reviews.llvm.org/D71213)
changed the signature of CreateMemSet() function, which causes the following
compilation errors:

...
src/ast/irbuilderbpf.cpp:85:52: error: no matching function for call to 'bpftrace::ast::IRBuilderBPF::CreateMemSet'
   85 |     CreateMemSet(alloca, getInt64(0), stype.size, 1);
      |
...

Use LLVM_VERSION_MAJOR macro to select the appropriate call.

Fixes: #1165
Signed-off-by: Ovidiu Panait <ovpanait@gmail.com>

fbs commented:
In irbuilder.h we've defined a few CREATE_MEMCPY macros to handle LLVM version specific things. Doing it like that avoids having a ton of ifdefs throughout the code.

Also, I thought that llvm deprecated the legacy pass manager with version 10, or does it still work? :)

ovpanait commented:
Hi,

Thanks for the feedback! I have updated the patch and added the CREATE_MEMSET macro in src/ast/irbuilderbpf.h.

As for the the legacy pass manager deprecation, sorry but I am not really familiar with the llvm api. On my side it builds fine with these changes only.

danobi commented:
Please see the clang-format failures in the build. Otherwise LGTM

ovpanait commented:
Updated the patch as per clang-format test indication.

fbs commented:
nice. Thanks!
